message RegisterDescription { 
  required int32 register_number = 1;
  required string register_name = 2;
}

message RegisterMap {
  repeated RegisterDescription register_map = 1;
}

message RegisterDelta {
  required  int32 register_number = 1;
  required  uint64 register_value = 2;
}

message TraceEvent {
  required int64          n = 1;
  required bytes          instruction = 2;
  repeated RegisterDelta  regs = 3;
  required int32          status = 4;
}

message TraceHeader {
  required string path = 1;
  repeated string cmdline = 2;
  required uint32 machine_width = 3;
}
